### PR TITLE
Add readonly modifier to internal static members

### PR DESCRIPTION
**Pull request by xtqqczze**
# PR Summary

* Add readonly modifier to internal static members.

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [x] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and enumerated concerns in the summary. This may include:
        - Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode (which runs in a different PS Host).
        - Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
        - Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
        - Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).

### CODE REVIEW

**Comment by powercode** [comment]
Why readonly instead of const?

**Comment by xtqqczze** [comment]
@powercode I considered a change to `const` would have uncertain benefit, and might be a risk, because all variables which consume const would then be hardcoded. 

In theory since the fields are `internal` there should no danger in using `const`, but I was not confident there could be no issues.

https://blogs.msdn.microsoft.com/armenk/2013/03/01/how-dangerous-can-be-public-const-while-upgrading-or-fixing-a-component/

**Comment by xtqqczze** [comment]
rebased to resolve merge conflicts

**Comment by xtqqczze** [comment]
there are no major conflicts with #11916

**Comment by xtqqczze** [comment]
@iSazonov I have made amendments, can you review?

File: src/Microsoft.Management.Infrastructure.CimCmdlets/CimSessionOperations.cs [hunk 1]
```diff
@@ -141,19 +141,19 @@ internal PSObject GetPSObject()
             if (psObject == null)
             {
                 psObject = new PSObject(this.cimSession);
-                psObject.Properties.Add(new PSNoteProperty(CimSessionState.idPropName, this.sessionId));
-                psObject.Properties.Add(new PSNoteProperty(CimSessionState.namePropName, this.name));
-                psObject.Properties.Add(new PSNoteProperty(CimSessionState.instanceidPropName, this.instanceId));
-                psObject.Properties.Add(new PSNoteProperty(CimSessionState.computernamePropName, this.ComputerName));
-                psObject.Properties.Add(new PSNoteProperty(CimSessionState.protocolPropName, this.Protocol));
+                psObject.Properties.Add(new PSNoteProperty(CimSessionState.IdPropName, this.sessionId));
```
Comment (iSazonov) [inline src/Microsoft.Management.Infrastructure.CimCmdlets/CimSessionOperations.cs:144]: It is not follow our code conventions https://github.com/PowerShell/PowerShell/blob/master/docs/dev-process/coding-guidelines.md#naming-conventions

It should be `s_idPropName`.

But the change is not related the PR. Please revert name changes and leave only adding `readonly`

**Comment by xtqqczze** [comment]
@iSazonov force pushed to remove name changes, but now we have StyleCop rule violations, [SA1304NonPrivateReadonlyFieldsMustBeginWithUpperCaseLetter](https://github.com/DotNetAnalyzers/StyleCopAnalyzers/blob/master/documentation/SA1304.md)

**Review by iSazonov — APPROVED** [review]

**Comment by xtqqczze** [comment]
rebased to resolve merge conflicts

**Comment by iSazonov** [comment]
> force pushed to remove name changes, but now we have StyleCop rule violations, SA1304NonPrivateReadonlyFieldsMustBeginWithUpperCaseLetter

Our code convention is :
> Use _camelCase to name internal and private fields and use readonly where possible.

So we have a conflict with StyleCop rule. But we can not use the rule without changing our code convention.
/cc @daxian-dbw for conclusion.

**Comment by xtqqczze** [comment]
> So we have a conflict with StyleCop rule. But we can not use the rule without changing our code convention.
> /cc @daxian-dbw for conclusion.

@iSazonov I have opened #12819 so we can make a general decision on the code convention.

**Comment by xtqqczze** [comment]
@iSazonov I am not sure why CodeFactor finds `NonPrivateReadonlyFieldsMustBeginWithUpperCaseLetter` issues as the rule is not enabled in `setting.stylecop` or the [default rule-set](https://github.com/codefactor-io/default-configs/blob/master/Settings.StyleCop).

**Comment by iSazonov** [comment]
@xtqqczze Perhaps it is CodeFactor effiect.

**Comment by xtqqczze** [comment]
> 
> 
> @xtqqczze Perhaps it is CodeFactor effiect.

@iSazonov The rule is not present in the CodeFactor [default rule-set](https://github.com/codefactor-io/default-configs/blob/master/Settings.StyleCop).

**Comment by xtqqczze** [comment]
@iSazonov I have PR #12855 to disable `NonPrivateReadonlyFieldsMustBeginWithUpperCaseLetter ` in `Settings.StyleCop`.
